src/core/auth/wallet-authenticator.ts

```diff
@@ -0,0 +1,342 @@
+import { ethers } from "ethers";
+import { isBrowser } from "../../common/utils";
+import { SDKOptions } from "../../schema";
+import {
+  LoginOptions,
+  LoginPayload,
+  AuthenticationOptions,
+  LoginPayloadData,
+  LoginPayloadDataSchema,
+  AuthenticationPayloadDataSchema,
+  AuthenticationPayloadData,
+  LoginOptionsSchema,
+  VerifyOptionsSchema,
+  VerifyOptions,
+  AuthenticationOptionsSchema,
+} from "../../schema/auth";
+import { RPCConnectionHandler } from "../classes/rpc-connection-handler";
+import { NetworkOrSignerOrProvider } from "../types";
+import { UserWallet } from "../wallet";
+
+/**
+ * Wallet Authenticator
+ * @remarks The wallet authenticator enables server-side applications to securely identify the
+ * connected wallet address of users on the client-side, and also enables users to authenticate
+ * to any backend using just their wallet. It implements the JSON Web Token (JWT) authentication
+ * standard.
+ *
+ * @example
+ * ```javascript
+ * // We specify the domain of the application to authenticate to
+ * const domain = "thirdweb.com"
+ *
+ * // On the client side, we can generate a payload for the connected wallet to login
+ * const loginPayload = await sdk.auth.login(domain);
+ *
+ * // Then on the server side, we can securely verify the connected client-side address
+ * const address = sdk.auth.verify(domain, loginPayload);
+ *
+ * // And we can also generate an authentication token to send to the client
+ * const token = sdk.auth.generate(domain, loginPayload);
+ *
+ * // Finally, the token can be send from the client to the server to make authenticated requests
+ * // And the server can use the following function to authenticate a token and verify the associated address
+ * const address = sdk.auth.authenticate(domain, token);
+ * ```
+ * @beta
+ */
+export class WalletAuthenticator extends RPCConnectionHandler {
+  private wallet: UserWallet;
+
+  constructor(
+    network: NetworkOrSignerOrProvider,
+    wallet: UserWallet,
+    options: SDKOptions,
+  ) {
+    super(network, options);
+    this.wallet = wallet;
+  }
+
+  /**
+   * Login With Connected Wallet
+   * @remarks Client-side function that allows the connected wallet to login to a server-side application.
+   * Generates a login payload that can be sent to the server-side for verification or authentication.
+   *
+   * @param domain - The domain of the server-side application to login to
+   * @param options - Optional configuration options for the login request
+   * @returns Login payload that can be used on the server-side to verify the login request or authenticate
+   *
+   * @example
+   * ```javascript
+   * // Add the domain of the application users will login to, this will be used throughout the login process
+   * const domain = "thirdweb.com";
+   * // Generate a signed login payload for the connected wallet to authenticate with
+   * const loginPayload = await sdk.auth.login(domain);
+   * ```
+   */
+  public async login(
```

Comment:
does the doc show this as @beta as well?

Reply:
Yep looks like all get marked as @beta in docs